vumi/components/message_store.py

```diff
@@ -124,18 +154,24 @@ class InboundMessage(Model):
     # Extra fields for compound indexes
     batches_with_timestamps = ListOf(Unicode(), index=True)
     batches_with_addresses = ListOf(Unicode(), index=True)
+    batches_with_addresses_reverse = ListOf(Unicode(), index=True)
 
     def save(self):
         # We override this method to set our index fields before saving.
-        batches_with_timestamps = []
-        batches_with_addresses = []
+        self.batches_with_timestamps = []
+        self.batches_with_addresses = []
+        self.batches_with_addresses_reverse = []
         timestamp = self.msg['timestamp']
+        if not isinstance(timestamp, basestring):
+            timestamp = format_vumi_date(timestamp)
+        reverse_ts = to_reverse_timestamp(timestamp)
         for batch_id in self.batches.keys():
-            batches_with_timestamps.append(u"%s$%s" % (batch_id, timestamp))
-            batches_with_addresses.append(
+            self.batches_with_timestamps.append(
+                u"%s$%s" % (batch_id, timestamp))
+            self.batches_with_addresses.append(
                 u"%s$%s$%s" % (batch_id, timestamp, self.msg['from_addr']))
-        self.batches_with_timestamps = batches_with_timestamps
-        self.batches_with_addresses = batches_with_addresses
+            self.batches_with_addresses_reverse.append(
+                u"%s$%s$%s" % (batch_id, reverse_ts, self.msg['from_addr']))
```

Comment:
Probably not worth doing much work on here since we are going to be using github.com/praekelt/vumi-message-store some time in the future, but I'm seeing quite a bit of repitition between `InboundMessage.save` and `OutboundMessage.save`. Would it maybe be worth removing that repitition in vumi-message-store, if these models were copied across as is?